Amazon.IonObjectMapper/IonSerializer.cs

```diff
@@ -504,5 +563,20 @@ private IIonSerializer GetPrimitiveSerializer<T>(IIonSerializer defaultSerialize
 
             return defaultSerializer;
         }
+
+        private IIonSerializer CreateCustomSerializer (Type targetType)
+        {
+            var customSerializerAttribute = targetType.GetCustomAttribute<IonSerializerAttribute>();
+            if (customSerializerAttribute.Factory != null) 
+            {
+                return ((IIonSerializerFactory)Activator.CreateInstance(customSerializerAttribute.Factory)).Create(options, options.CustomContext);

```

Comment:
I think splitting this up into two lines (create factory, then create serializer from factory) would make it easier to follow.